# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsIContentInlines.h
# Commit: 29195ee87cf4
# Full Hash: 29195ee87cf436e0b76854ec86c114c1d1ced8fe
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-10 09:29:34
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231588
# ==============================================================================

diff -r a86c7713f936 -r 29195ee87cf4 dom/base/nsIContentInlines.h
--- a/dom/base/nsIContentInlines.h	Fri May 09 17:16:36 2025 +0000
+++ b/dom/base/nsIContentInlines.h	Fri May 09 17:16:36 2025 +0000
@@ -144,9 +144,8 @@
   return ::GetFlattenedTreeParentNode<nsINode::eForStyle>(this);
 }
 
-inline nsIContent* nsINode::GetFlattenedTreeParentNodeForSelection() const {
-  nsINode* parent = ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
-  return (parent && parent->IsContent()) ? parent->AsContent() : nullptr;
+inline nsINode* nsINode::GetFlattenedTreeParentNodeForSelection() const {
+  return ::GetFlattenedTreeParentNode<nsINode::eForSelection>(this);
 }
 
 inline bool nsINode::NodeOrAncestorHasDirAuto() const {